### PR TITLE
feat: add event.senderFrame property returning the webFrameMain

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Also add missing `event.processId`, which needs to be passed to `webFrameMain.fromId()`.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes
Notes: Added `event.senderFrame` property returning the originating `webFrameMain` of the IPC message.

### CODE REVIEW

File: docs/api/structures/ipc-main-event.md [hunk 1]
```diff
@@ -1,8 +1,10 @@
 # IpcMainEvent Object extends `Event`
 
 * `frameId` Integer - The ID of the renderer frame that sent this message
+* `processId` Integer - The ID of the process which owns the frame
```
Comment (nornagon) [inline docs/api/structures/ipc-main-event.md:4]: Is this the OS process ID or the internal process ID? The docs should say.

File: docs/api/web-frame-main.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ win.webContents.on(
 )
 ```
 
-You can also access frames of existing pages by using the `webFrame` property
+You can also access frames of existing pages by using the `mainFrame` property
```
Comment (nornagon) [inline docs/api/web-frame-main.md:29]: should this fix be a separate PR + backported to 12.x?

File: lib/browser/api/web-contents.ts [hunk 3]
```diff
@@ -493,6 +493,12 @@ const addReplyInternalToEvent = (event: any) => {
   });
 };
 
+const addSenderFrameToEvent = (event: any) => {
+  Object.defineProperty(event, 'senderFrame', {
+    get: () => webFrameMain.fromId(event.processId, event.frameId)
+  });
```
Comment (nornagon) [inline lib/browser/api/web-contents.ts:499]: I'd prefer this to be resistant to the following:

```js
wc.on('event', (event) => {
  event.processId = 123
  event.senderFrame // ==> returns some other frame because the process ID changed
})
```

```suggestion
  const { processId, frameId } = event
  Object.defineProperty(event, 'senderFrame', {
    get: () => webFrameMain.fromId(processId, frameId)
  });
```

**Review by nornagon — CHANGES_REQUESTED** [review]

File: docs/api/web-frame-main.md (hunk 2 above)
Comment (miniak) [reply docs/api/web-frame-main.md:29]: I would like to backport this whole PR to 12-x-y if possible

**Review by nornagon — APPROVED** [review]
API LGTM

File: lib/browser/api/web-contents.ts [hunk 4]
```diff
@@ -493,6 +493,13 @@ const addReplyInternalToEvent = (event: any) => {
   });
 };
 
+const addSenderFrameToEvent = (event: any) => {
+  const { processId, frameId } = event;
+  Object.defineProperty(event, 'senderFrame', {
+    get: () => webFrameMain.fromId(processId, frameId)
+  });
+};
```
Comment (samuelmaddock) [inline lib/browser/api/web-contents.ts:501]: This is great!

**Review by samuelmaddock — APPROVED** [review]
The getter is going to be very handy. Nice!

**Review by jkleinsc — APPROVED** [review]
API LGTM as well

**Comment by nornagon** [comment]
looks like tests are failing. btw `processId` is also added by https://github.com/electron/electron/pull/26875

**Comment by miniak** [comment]
@nornagon I've noticed that, I had it originally
```cpp
    dict.Set("processId", frame->GetProcess()->GetID());
    dict.Set("frameId", frame->GetRoutingID());
```
and then changed it to match your PR to avoid conflict
```cpp
    dict.Set("frameId", frame->GetRoutingID());
    dict.Set("processId", frame->GetProcess()->GetID());
```
which broke the snapshot for tests

**Comment by nornagon** [comment]
@miniak the linked PR updates the tests to replace the process id with a placeholder in the snapshot.

**Review by nornagon — APPROVED** [review]
